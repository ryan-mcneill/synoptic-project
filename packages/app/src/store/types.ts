// Constants

export const FETCH_SONGS = "FETCH_SONGS";
export const FETCH_SONGS_FAILURE = "FETCH_SONGS_FAILURE";
export const FETCH_SONGS_SUCCESS = "FETCH_SONGS_SUCCESS";

export const FETCH_ARTISTS = "FETCH_ARTISTS";
export const FETCH_ARTISTS_FAILURE = "FETCH_ARTISTS_FAILURE";
export const FETCH_ARTISTS_SUCCESS = "FETCH_ARTISTS_SUCCESS";

export const FETCH_ALBUMS = "FETCH_ALBUMS";
export const FETCH_ALBUMS_FAILURE = "FETCH_ALBUMS_FAILURE";
export const FETCH_ALBUMS_SUCCESS = "FETCH_ALBUMS_SUCCESS";

export const FETCH_PLAYLISTS = "FETCH_PLAYLISTS";
export const FETCH_PLAYLISTS_FAILURE = "FETCH_PLAYLISTS_FAILURE";
export const FETCH_PLAYLISTS_SUCCESS = "FETCH_PLAYLISTS_SUCCESS";

export const ADD_TO_PLAYLIST = "ADD_TO_PLAYLIST";
export const ADD_TO_PLAYLIST_FAILURE = "ADD_TO_PLAYLIST_FAILURE";
export const ADD_TO_PLAYLIST_SUCCESS = "ADD_TO_PLAYLIST_SUCCESS";

export const SET_PLAYLIST = "SET_PLAYLIST";
export const PLAYLIST_ADD = "PLAYLIST_ADD";
export const PLAYLIST_REMOVE = "PLAYLIST_REMOVE";
export const UPDATE_PLAYLIST = "UPDATE_PLAYLIST";

export const SET_SONG = "SET_SONG";
export const SET_BACKGROUND = "SET_BACKGROUND";
export const TOGGLE_PLAYING = "TOGGLE_PLAYING";
export const NEXT_SONG = "NEXT_SONG";
export const PREV_SONG = "PREV_SONG";
export const SHUFFLE_SONGS = "SHUFFLE_SONGS";

export const SET_IS_OPEN = "SET_IS_OPEN";
export const SET_SELECTED_TAB = "SET_SELECTED_TAB";

export const TIMED_OUT = "TIMED_OUT";

// Typescript types

export interface BasicDetails {
  _id: string;
  name: string;
}

export interface Song extends BasicDetails {
  artists: BasicDetails[];
  albums: BasicDetails[];
}

export interface Background {
  colour: string;
  isDark: boolean;
}

export interface CurrentSongState {
  isPlaying: boolean;
  data?: Song;
  song?: string;
  albumArt?: string;
  background?: Background;
}

export interface PlaylistState {
  previous: Song[];
  current?: Song;
  next: Song[];
}

export interface ArtistsAndAlbumsData {
  [key: string]: {
    name: string;
    songs: Song[];
  };
}

export interface ArtistsAndAlbumsState {
  error: any;
  loading: boolean;
  data?: ArtistsAndAlbumsData;
}

export type Tabs = "PLAYLIST" | "SONGS" | "SEARCH";

export interface MenuState {
  isOpen: boolean;
  selectedTab: Tabs;
}

export interface State {
  currentSong: CurrentSongState;
  playlist: PlaylistState;
  artists: ArtistsAndAlbumsState;
  albums: ArtistsAndAlbumsState;
  playlists: ArtistsAndAlbumsState;
  menu: MenuState;
  songs: {
    error: any;
    loading: boolean;
    data?: Song[];
  };
  timedOut: boolean;
}

// Action typescript types

export interface SetPlaylistAction {
  type: typeof SET_PLAYLIST;
  payload: {
    firstSong: Song;
    nextSongs?: Song[];
  };
}

export interface PlaylistAddAction {
  type: typeof PLAYLIST_ADD;
  payload: Song;
}

export interface PlaylistRemoveAction {
  type: typeof PLAYLIST_REMOVE;
  payload: Song;
}

export interface UpdatePlaylistAction {
  type: typeof UPDATE_PLAYLIST;
  payload: {
    previous?: Song[];
    current?: Song;
    next?: Song[];
  };
}

export type PlaylistActions =
  | UpdatePlaylistAction
  | SetPlaylistAction
  | PlaylistAddAction
  | PlaylistRemoveAction;

export interface FetchSongsAction {
  type: typeof FETCH_SONGS;
}

export interface FetchSongsFailureAction {
  type: typeof FETCH_SONGS_FAILURE;
  payload: {
    error: Error;
  };
}

export interface FetchSongsSuccessAction {
  type: typeof FETCH_SONGS_SUCCESS;
  payload: Song[];
}

export type SongsActions =
  | FetchSongsAction
  | FetchSongsFailureAction
  | FetchSongsSuccessAction;

export interface FetchArtistsAction {
  type: typeof FETCH_ARTISTS;
}

export interface FetchArtistsFailureAction {
  type: typeof FETCH_ARTISTS_FAILURE;
  payload: {
    error: Error;
  };
}

export interface FetchArtistsSuccessAction {
  type: typeof FETCH_ARTISTS_SUCCESS;
  payload: Song[];
}

export type ArtistsActions =
  | FetchArtistsAction
  | FetchArtistsFailureAction
  | FetchArtistsSuccessAction;

export interface FetchAlbumsAction {
  type: typeof FETCH_ALBUMS;
}

export interface FetchAlbumsFailureAction {
  type: typeof FETCH_ALBUMS_FAILURE;
  payload: {
    error: Error;
  };
}

export interface FetchAlbumsSuccessAction {
  type: typeof FETCH_ALBUMS_SUCCESS;
  payload: Song[];
}

export type AlbumsActions =
  | FetchAlbumsAction
  | FetchAlbumsFailureAction
  | FetchAlbumsSuccessAction;

export interface FetchPlaylistsAction {
  type: typeof FETCH_PLAYLISTS;
}

export interface FetchPlaylistsFailureAction {
  type: typeof FETCH_PLAYLISTS_FAILURE;
  payload: {
    error: Error;
  };
}

export interface FetchPlaylistsSuccessAction {
  type: typeof FETCH_PLAYLISTS_SUCCESS;
  payload: Song[];
}

export interface AddToPlaylistAction {
  type: typeof ADD_TO_PLAYLIST;
}

export interface AddToPlaylistFailureAction {
  type: typeof ADD_TO_PLAYLIST_FAILURE;
  payload: {
    error: Error;
  };
}

export interface AddToPlaylistSuccessAction {
  type: typeof ADD_TO_PLAYLIST_SUCCESS;
  payload: Song[];
}

export type PlaylistsActions =
  | FetchPlaylistsAction
  | FetchPlaylistsFailureAction
  | FetchPlaylistsSuccessAction
  | AddToPlaylistAction
  | AddToPlaylistFailureAction
  | AddToPlaylistSuccessAction;

export interface SetSongAction {
  type: typeof SET_SONG;
  data: Song;
}

export interface SetBackgroundAction {
  type: typeof SET_BACKGROUND;
  data: Background;
}

export interface TogglePlayingAction {
  type: typeof TOGGLE_PLAYING;
  isPlaying?: boolean;
}

export interface NextSongAction {
  type: typeof NEXT_SONG;
}

export interface PrevSongAction {
  type: typeof PREV_SONG;
}

export interface ShuffleSongsAction {
  type: typeof SHUFFLE_SONGS;
}

export type CurrentSongActions =
  | SetSongAction
  | SetBackgroundAction
  | TogglePlayingAction
  | NextSongAction
  | PrevSongAction;

export type ChangeSongType = "PREV" | "NEXT";

export interface SetIsOpenAction {
  type: typeof SET_IS_OPEN;
  isOpen?: boolean;
}

export interface SetSelectedTab {
  type: typeof SET_SELECTED_TAB;
  tab: Tabs;
}

export type MenuActions = SetIsOpenAction | SetSelectedTab;

export interface TimedOutAction {
  type: typeof TIMED_OUT;
  hasTimedOut: boolean;
}
