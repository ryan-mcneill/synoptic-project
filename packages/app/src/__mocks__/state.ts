export const mockState = {
  albums: {
    loading: false,
    data: {
      "5f1423c54ee369215e05625e": {
        name: "Sample Album A",
        songs: [
          {
            _id: "5f14218d4ee369215e05624f",
            name: "Hey",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056256",
            name: "Cute",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056257",
            name: "Going Higher",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      },
      "5f1423c54ee369215e05625f": {
        name: "Sample Album B",
        songs: [
          {
            _id: "5f1422054ee369215e056255",
            name: "A New Beginning",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          }
        ]
      }
    },
    error: null
  },
  artists: {
    loading: false,
    data: {
      "5f14228b4ee369215e05625a": {
        name: "Benjamin Tissot",
        songs: [
          {
            _id: "5f14218d4ee369215e05624f",
            name: "Hey",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056256",
            name: "Cute",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056257",
            name: "Going Higher",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      },
      "5f14228b4ee369215e05625b": {
        name: "BenSound",
        songs: [
          {
            _id: "5f1422054ee369215e056255",
            name: "A New Beginning",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      }
    },
    error: null
  },
  currentSong: {
    isPlaying: false,
    background: {
      isDark: false,
      colour: "#e0c1e2"
    }
  },
  menu: {
    isOpen: false,
    selectedTab: "PLAYLIST"
  },
  playlist: {
    previous: [],
    current: {
      _id: "5f14218d4ee369215e05624f",
      name: "Hey",
      artists: [
        {
          _id: "5f14228b4ee369215e05625a",
          name: "Benjamin Tissot"
        }
      ],
      albums: [
        {
          _id: "5f1423c54ee369215e05625e",
          name: "Sample Album A"
        }
      ]
    },
    next: [
      {
        _id: "5f1422054ee369215e056256",
        name: "Cute",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056257",
        name: "Going Higher",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1ad1ef17e08f691bb74048",
        name: "Creative Minds",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          },
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      }
    ]
  },
  playlists: {
    loading: false,
    data: {
      "5f1ffca0faf3a97662868ac0": {
        name: "Test Playlist 1",
        songs: [
          {
            _id: "5f1422054ee369215e056257",
            name: "Going Higher",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056255",
            name: "A New Beginning",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          }
        ]
      },
      "5f21389bc969de0689f10656": {
        name: "Test Playlist 2",
        songs: [
          {
            _id: "5f14218d4ee369215e05624f",
            name: "Hey",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056256",
            name: "Cute",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      },
      "5f213936c969de0689f10657": {
        name: "Test Playlist 3",
        songs: [
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          }
        ]
      }
    },
    error: null
  },
  songs: {
    loading: false,
    data: [
      {
        _id: "5f1422054ee369215e056255",
        name: "A New Beginning",
        artists: [
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625f",
            name: "Sample Album B"
          }
        ]
      },
      {
        _id: "5f1ad1ef17e08f691bb74048",
        name: "Creative Minds",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          },
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056256",
        name: "Cute",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056257",
        name: "Going Higher",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f14218d4ee369215e05624f",
        name: "Hey",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056258",
        name: "Ukulele",
        artists: [
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625f",
            name: "Sample Album B"
          }
        ]
      }
    ],
    error: null
  }
};
