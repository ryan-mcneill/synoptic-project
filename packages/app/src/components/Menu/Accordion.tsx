import React, { FC, ReactElement, useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronDown, faChevronUp } from "@fortawesome/free-solid-svg-icons";
import { SongList } from "./";
import { Song } from "../../store/types";

interface AccordionProps {
  title: string;
  data: Song[];
}

const Accordion: FC<AccordionProps> = ({
  title,
  data
}): ReactElement => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <>
      <div
        style={{
          width: "100%",
          display: "flex",
          alignItems: "center",
          fontSize: "48px",
          padding: "1.5vh 0"
        }}
        onClick={() => setIsOpen(!isOpen)}
      >
        <div style={{ width: "calc(100% - 42px)" }}>{title}</div>
        <div style={{ width: "42px" }}>
          <FontAwesomeIcon icon={isOpen ? faChevronUp : faChevronDown} />
        </div>
      </div>

      <SongList isVisible={isOpen} data={data} />
    </>
  );
};

export default Accordion;
