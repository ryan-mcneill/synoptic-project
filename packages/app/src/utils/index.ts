import { artistsToString } from "./artistsToString";
import { filterData } from "./filterData";
import { filterSongs } from "./filterSongs";
import { playlistsToArray } from "./playlistsToArray";

export { artistsToString, filterData, filterSongs, playlistsToArray };
