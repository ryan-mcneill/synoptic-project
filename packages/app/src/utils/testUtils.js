import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import reducers from "../store/reducers";

function render(
  ui,
  {
    initialState,
    store = createStore(reducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
