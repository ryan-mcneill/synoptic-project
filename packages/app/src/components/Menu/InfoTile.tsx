import React, { FC, ReactElement, useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronCircleRight } from "@fortawesome/free-solid-svg-icons";
import { SongMenu } from "./";
import { Song } from "../../store/types";
import { artistsToString } from "../../utils";

interface InfoTile {
  selected?: boolean;
  onClick?: () => void;
  hasMenu?: boolean;
  data: Song;
}

const InfoTile: FC<InfoTile> = ({
  data,
  onClick,
  selected = false,
  hasMenu = true
}): ReactElement => {
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  const artists = data.artists ? artistsToString(data.artists) : "";

  const handleClick = () => {
    setIsMenuOpen(true);
    onClick && onClick();
  };

  return (
    <div>
      <div
        style={{
          width: "100%",
          display: "flex",
          alignItems: "center",
          fontSize: "48px",
          padding: "2vh 0"
        }}
        onClick={handleClick}
      >
        <div style={{ width: "57px", height: "auto", paddingRight: "32px" }}>
          {selected && <FontAwesomeIcon icon={faChevronCircleRight} />}
        </div>
        <div>
          <div style={{ fontSize: "48px" }}>{data.name}</div>
          {artists !== "" && <div style={{ fontSize: "36px" }}>{artists}</div>}
        </div>
      </div>
      {hasMenu && isMenuOpen && (
        <SongMenu
          artists={artists}
          data={data}
          onClose={() => setIsMenuOpen(false)}
        />
      )}
    </div>
  );
};

export default InfoTile;
