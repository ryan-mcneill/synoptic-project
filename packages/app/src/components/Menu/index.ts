import Accordion from "./Accordion";
import InfoTile from "./InfoTile";
import PlaylistTab from "./PlaylistTab";
import SearchTab from "./SearchTab";
import SongList from "./SongList";
import SongMenu from "./SongMenu";
import SongsTab from "./SongsTab";
import TabButton from "./TabButton";
import Tabs from "./Tabs";

export {
  Accordion,
  InfoTile,
  PlaylistTab,
  SearchTab,
  SongList,
  SongMenu,
  SongsTab,
  TabButton,
  Tabs
};
