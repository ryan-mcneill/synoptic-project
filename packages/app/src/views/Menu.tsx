import React, { FC, ReactElement } from "react";
import { Action } from "redux";
import { connect, ConnectedProps } from "react-redux";
import { ThunkDispatch } from "redux-thunk";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEllipsisH } from "@fortawesome/free-solid-svg-icons";
import { Tabs } from "../components/Menu";
import { setIsOpen } from "../store/actions";
import { State } from "../store/types";

const mapDispatchToProps = (dispatch: ThunkDispatch<State, void, Action>) => ({
  setIsOpen: (isOpen?: boolean) => dispatch(setIsOpen(isOpen))
});

const mapStateToProps = (state: State) => ({
  albumsLoading: state.albums.loading,
  artistsLoading: state.artists.loading,
  isOpen: state.menu.isOpen
});

const connector = connect(mapStateToProps, mapDispatchToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

const Menu: FC<PropsFromRedux> = ({
  albumsLoading,
  artistsLoading,
  isOpen,
  setIsOpen
}): ReactElement =>
  !albumsLoading && !artistsLoading ? (
    <div
      style={{
        position: "absolute",
        bottom: isOpen ? 0 : "-90vh",
        transition: "all 0.5s ease"
      }}
    >
      <div
        style={{
          textAlign: "center",
          fontSize: "72px",
          position: "relative",
          left: "50%",
          bottom: "-1px",
          transform: "translateX(-50%)",
          height: "72px",
          width: "128px",
          backgroundColor: "rgba(255, 255, 255, .5)",
          borderRadius: "24px 24px 0 0",
          backdropFilter: "blur(20px)"
        }}
        onClick={() => setIsOpen()}
      >
        <FontAwesomeIcon icon={faEllipsisH} />
      </div>
      <Tabs />
    </div>
  ) : (
    <></>
  );

export default connector(Menu);
