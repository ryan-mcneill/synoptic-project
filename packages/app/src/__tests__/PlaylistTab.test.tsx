import React from "react";
import { PlaylistTab } from "../components/Menu";
import { render } from "../utils/testUtils";
import { mockState } from "../__mocks__/state";
import { PlaylistState } from "../store/types";

describe("The PlaylistTab component", () => {
  let initialState: { playlist: PlaylistState } = {
    playlist: {
      previous: [],
      next: []
    }
  };

  const song = mockState.playlist.current;

  beforeEach(() => {
    initialState = {
      playlist: {
        previous: [],
        next: []
      }
    };
  });

  it("renders correctly with no songs playing", () => {
    const { container } = render(<PlaylistTab />, {
      initialState
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with songs in previous state", () => {
    initialState.playlist.previous = mockState.playlist.next;
    const { container } = render(<PlaylistTab />, {
      initialState
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with songs in current state", () => {
    initialState.playlist.current = song;
    const { container } = render(<PlaylistTab />, {
      initialState
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with songs in next state", () => {
    initialState.playlist.next = mockState.playlist.next;
    const { container } = render(<PlaylistTab />, {
      initialState
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
