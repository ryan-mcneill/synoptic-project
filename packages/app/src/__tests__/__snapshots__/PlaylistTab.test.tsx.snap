// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PlaylistTab component renders correctly with no songs playing 1`] = `
<div>
  <div
    style="font-size: 48px; text-align: center;"
  >
    Use the Songs or Search tab to add songs to the playlist.
  </div>
  <div
    style="display: flex; height: 4vh; width: 80vw; font-size: 36px; position: absolute; bottom: 5vh;"
  >
    <input
      disabled=""
      style="width: 50%; font-size: 36px; font-weight: bolder; padding-left: 2vw; background: none;"
      type="button"
      value="CLEAR"
    />
    <input
      disabled=""
      style="width: 50%; font-size: 36px; font-weight: bolder; padding-left: 2vw; background: none;"
      type="button"
      value="SHUFFLE"
    />
  </div>
</div>
`;

exports[`The PlaylistTab component renders correctly with songs in current state 1`] = `
<div>
  <div>
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
    >
      <div
        style="width: 57px; height: auto; padding-right: 32px;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-circle-right fa-w-16 "
          data-icon="chevron-circle-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div>
        <div
          style="font-size: 48px;"
        >
          Hey
        </div>
        <div
          style="font-size: 36px;"
        >
          Benjamin Tissot
        </div>
      </div>
    </div>
  </div>
  <div
    style="display: flex; height: 4vh; width: 80vw; font-size: 36px; position: absolute; bottom: 5vh;"
  >
    <input
      style="width: 50%; font-size: 36px; font-weight: bolder; padding-left: 2vw; background: none;"
      type="button"
      value="CLEAR"
    />
    <input
      style="width: 50%; font-size: 36px; font-weight: bolder; padding-left: 2vw; background: none;"
      type="button"
      value="SHUFFLE"
    />
  </div>
</div>
`;

exports[`The PlaylistTab component renders correctly with songs in next state 1`] = `
<div>
  <div>
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
    >
      <div
        style="width: 57px; height: auto; padding-right: 32px;"
      />
      <div>
        <div
          style="font-size: 48px;"
        >
          Cute
        </div>
        <div
          style="font-size: 36px;"
        >
          Benjamin Tissot
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
    >
      <div
        style="width: 57px; height: auto; padding-right: 32px;"
      />
      <div>
        <div
          style="font-size: 48px;"
        >
          Going Higher
        </div>
        <div
          style="font-size: 36px;"
        >
          Benjamin Tissot
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
    >
      <div
        style="width: 57px; height: auto; padding-right: 32px;"
      />
      <div>
        <div
          style="font-size: 48px;"
        >
          Creative Minds
        </div>
        <div
          style="font-size: 36px;"
        >
          Benjamin Tissot, BenSound
        </div>
      </div>
    </div>
  </div>
  <div
    style="display: flex; height: 4vh; width: 80vw; font-size: 36px; position: absolute; bottom: 5vh;"
  >
    <input
      disabled=""
      style="width: 50%; font-size: 36px; font-weight: bolder; padding-left: 2vw; background: none;"
      type="button"
      value="CLEAR"
    />
    <input
      disabled=""
      style="width: 50%; font-size: 36px; font-weight: bolder; padding-left: 2vw; background: none;"
      type="button"
      value="SHUFFLE"
    />
  </div>
</div>
`;

exports[`The PlaylistTab component renders correctly with songs in previous state 1`] = `
<div>
  <div>
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
    >
      <div
        style="width: 57px; height: auto; padding-right: 32px;"
      />
      <div>
        <div
          style="font-size: 48px;"
        >
          Cute
        </div>
        <div
          style="font-size: 36px;"
        >
          Benjamin Tissot
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
    >
      <div
        style="width: 57px; height: auto; padding-right: 32px;"
      />
      <div>
        <div
          style="font-size: 48px;"
        >
          Going Higher
        </div>
        <div
          style="font-size: 36px;"
        >
          Benjamin Tissot
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      style="width: 100%; display: flex; align-items: center; font-size: 48px; padding: 2vh 0px;"
    >
      <div
        style="width: 57px; height: auto; padding-right: 32px;"
      />
      <div>
        <div
          style="font-size: 48px;"
        >
          Creative Minds
        </div>
        <div
          style="font-size: 36px;"
        >
          Benjamin Tissot, BenSound
        </div>
      </div>
    </div>
  </div>
  <div
    style="display: flex; height: 4vh; width: 80vw; font-size: 36px; position: absolute; bottom: 5vh;"
  >
    <input
      disabled=""
      style="width: 50%; font-size: 36px; font-weight: bolder; padding-left: 2vw; background: none;"
      type="button"
      value="CLEAR"
    />
    <input
      disabled=""
      style="width: 50%; font-size: 36px; font-weight: bolder; padding-left: 2vw; background: none;"
      type="button"
      value="SHUFFLE"
    />
  </div>
</div>
`;
